src/support/serde.rs

```diff
@@ -171,12 +171,12 @@ mod tests {
             const LIMBS: usize = nlimbs(BITS);
             proptest!(|(value: Uint<BITS, LIMBS>)| {
                 let serialized = serde_json::to_string(&value).unwrap();
-                let deserialized = serde_json::from_str(&serialized).unwrap();
+                let deserialized = serde_json::from_str::<Uint<BITS, LIMBS>>(&serialized).unwrap();
```

Comment:
so to check my understanding, inference changed because the `serde_json::from_str` (and the other deser methods) must produce some type `T` that the `assert_eq!` then accepts on the next line. This `T` now has multiple candidates, whereas previously it had only 1 candidate (`Uint<Bits, Limbs>`), right?

I think this is acceptable breakage for our semver policy, as it is rare and can be trivially fixed by adding a type annotation

Reply:
that's right, it is mentioned here as well https://std-dev-guide.rust-lang.org/breaking-changes/new-trait-impls.html#inference-breaks-when-a-second-generic-impl-is-introduced